polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -577,7 +577,13 @@ async fn determine_our_validators<Context>(
 fn declare_message(
 	state: &mut State,
 	version: CollationVersion,
-) -> Option<Versioned<protocol_v1::CollationProtocol, protocol_v2::CollationProtocol>> {
+) -> Option<
+	Versioned<
+		protocol_v1::CollationProtocol,
+		protocol_v2::CollationProtocol,
+		protocol_vstaging::CollationProtocol,
```

Comment:
Why?

Reply:
Both Validation and Collation protocol use the same templated structure, so to make the compilation pass I opted to defined the collation version as well.
```
pub enum Versioned<V1, V2, VStaging> {
	/// V1 type.
	V1(V1),
	/// V2 type.
	V2(V2),
	/// VStaging type.
	VStaging(VStaging),
}
```

However, the collation staging won't be advertised, so it is there to be make the compilation pass: https://github.com/paritytech/polkadot-sdk/pull/1178/files#diff-ccf6dac7c0f390c65d31b155e518051af3dea517b1722905ded32c5d02b88849R130.

Looking, to see if I can circumvent it somehow.

Reply:
Turns-out they weren't actually needed, cleaned it up here: https://github.com/paritytech/polkadot-sdk/pull/1178/commits/fc286188b61dfa351cf41b48318b77eba3512878